book/src/development/define_lints.md

```diff
@@ -0,0 +1,179 @@
+# Define New Lints
+
+The journey of any new Clippy lint begins with one step.
+And this first step is to define the lint and register it to the codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+## Name the Lint
+
+A wise software engineer once said:
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+
+Naming a lint is no exception.
+Therefore, in case of uncertainty about if the name you chose fits the lint,
+please consult our [lint naming guidelines][lint_naming] or ask a maintainer
+on the corresponding Github issue or our Zulip chat.
+
+For now, let's imagine that your fellow developers at work tend to define some of their
+functions with the name `foo` and forget to re-name it to a more meaningful name
+when they submit a pull request.
+
+`foo` is a highly non-descriptive name for a function, so we want to detect this
+bad naming and fix it early on in the development process.
+
+For this, we will create a lint that detects these `foo` functions and
+help our fellow developers correct this bad practice. Note that in Clippy,
+lints are generally written in snake cases.
+We can name this new lint `foo_functions`.
+
+## Add and Register the Lint
+
+Now that a name is chosen, we shall register `foo_functions` as a lint to the codebase.
+There are two ways to register a lint.
+
+### Standalone
+
+If you believe that this new lint is a standalone lint, you can run the following
+command in your Clippy project:
+
+```sh
+$ cargo dev new_lint --name=foo_functions --pass=early --category=pedantic
+```
+
+Note that if not provided, the `category` of this new lint will default to `nursery`.
+See Clippy's [lint groups](../lints.md) for more information on categories.
+
+The `cargo dev new_lint` command will create a new file: `clippy_lints/src/foo_functions.rs`
+as well as [register the lint](#lint-registration).
```

Comment:
It will actually also create a test file in `tests/ui/foo_functions.stderr`

Reply:
I ran the command and I think it only generates `tests/ui/foo_functions.rs` 🤔

Reply:
Sorry, that's what I meant, but this is only referencing the rs file :upside_down_face: